pkg/ansible/paramconv/paramconv.go

```diff
@@ -70,6 +70,49 @@ func ToCamel(s string) string {
 	return ret
 }
 
+// preprocessWordMapping() will check if value contains special words mapped or its plural in
+// wordMapping, then processes it such that ToSnake() can convert it to snake case.
+// If value contains special word, the character "_" is appended as a prefix and postfix
+// to the special word found. For example, if the input string is "egressIP",
+// which contains is a special word "IP", the function will return "egress_IP".
+// If the last character of the special word is an "s" (i.e plural of the word
+// found in wordMapping), it is considered a part of that word and will be capitalized.
+func preprocessWordMapping(value string) string {
+	var x int
+	var y string
+
+	for _, word := range wordMapping {
+		x = strings.Index(value, word)
+		y = word
+		if x >= 0 {
+			// The special non-plural word appears at the end of the string.
+			if (x + len(y) - 1) == len(value)-1 {
+				value = value[:x] + "_" + value[x:]
+			} else {
+				// The special plural word occurs at the end, start, or somewhere in the middle of value.
+				if value[x+len(y)] == 's' {
+					if x+len(y) == len(value)-1 {
+						value = value[:x] + "_" + value[x:(x+len(y))] + "S"
+					} else if x == 0 {
+						value = value[:(x+len(y))] + "S" + "_" + value[(x+len(y)+1):]
+					} else {
+						value = value[:x] + "_" + value[x:(x+len(y))] + "S" + "_" + value[(x+len(y)+1):]
+					}
+					// The special non-plural word occurs at the start or somewhere in the middle of value.
+				} else {
+					if x == 0 {
+						value = value[:(x+len(y))] + "_" + value[(x+len(y)):]
+					} else {
+						value = value[:x] + "_" + value[x:(x+len(y))] + "_" + value[(x+len(y)):]
+					}
+				}
+			}
+		}
+	}
```

Comment:
I was able to remove all the extra `else` statements. The tests pass. I'm not 100% sure if the logic is the same yet. 

```golang
    for _, word := range wordMapping {                                             
        idx = strings.Index(value, word)                                           
        if idx >= 0 {                                                              
            // The special non-plural word appears at the end of the string.       
            if (idx + len(word) - 1) == len(value)-1 {                             
                value = value[:idx] + "_" + value[idx:]                            
            } else if value[idx+len(word)] == 's' {                                
                // The special plural word occurs at the end, start, or somewhere in the middle of value.
                if idx+len(word) == len(value)-1 {                                 
                    value = value[:idx] + "_" + value[idx:(idx+len(word))] + "S" 
                } else if idx == 0 {                                               
                    value = value[:(idx+len(word))] + "S" + "_" + value[(idx+len(word)+1):]
                } else {                                                           
                    value = value[:idx] + "_" + value[idx:(idx+len(word))] + "S" + "_" + value[(idx+len(word)+1):]
                }                                                                  
            } else if idx == 0 {                                                   
                // The special non-plural word occurs at the start or somewhere in the middle of value.
                value = value[:(idx+len(word))] + "_" + value[(idx+len(word)):] 
            } else {                                                               
                value = value[:idx] + "_" + value[idx:(idx+len(word))] + "_" + value[(idx+len(word)):]
            }                                                                      
        }                                                                          
    } 
```

Reply:
I just checked the logic. It should work fine. let me update the code with this and run some tests.

Reply:
I have implemented this logic (Removed the unnecessary else{} and eliminated the use of one extra variable